### PR TITLE
Improve the welcoming experience

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

When Gaphor is started, it shows an empty model.

Issue Number: #1157 

### What is the new behavior?

When gaphor is started, we open with a "Greeter", where you can pick a recently opened file (and later a new file from a Template/Scaffold).

The experience is based on [GNOME Builder](https://wiki.gnome.org/Apps/Builder). You're welcomed with your recently opened files in big button-ish things. There is an option to start a new model or open an existing model.

- [x] Remove "Open" button and add an entry in the hamburger dropdown.
- [x] Do no longer replace the model in an "empty" window.
- [x] Show something useful when there are no recent files.
- [ ] Test experience on Windows and macOS.

NB. This PR does not contain model templates yet.

### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### Other information

Models are now displayed by their name, without extension.

image

The main window looks cleaner now:

image

### CODE REVIEW

**Review by danyeaw — APPROVED** [review]
I really like this @amolenaar, you blew me away, nice one!

I tested it on Windows as well 👍

File: gaphor/ui/mainwindow.py [hunk 1]
```diff
@@ -245,8 +212,8 @@ def set_title(self):
 
         if self.filename:
             p = Path(self.filename)
-            title = p.name
-            subtitle = str(p.parent).replace(HOME, "~")
+            title = p.stem
+            subtitle = str(p).replace(HOME, "~")
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/ui/mainwindow.py:216]: The same argument goes for unexperienced Linux/macOS users.

I think they'll figure it out. It's a lot shorter than `C:\Users\myusername` at least ;)

What would be the alternative? `Development/test/testmodel.gaphor` or start it with ellipses instead?

**Comment by danyeaw** [comment]
Ya, I think it is fine. I don't think there is really a good equivalent in Windows since I doubt most Windows users know what `%USERPROFILE%` is and that would be ugly to put as a path in the GUI. At least PowerShell understands ~, so maybe it is becoming more familiar. :+1:

Where are the recently used models stored?

**Comment by amolenaar** [comment]
> Where are the recently used models stored?

The recent files are stored by GTK somewhere. This way it integrates well with GNOME.

**Comment by danyeaw** [comment]
> The recent files are stored by GTK somewhere. This way it integrates well with GNOME.

Got it, looks like removing ~/.local/share/recently-used.xbel does the trick to reset it so that I can see the welcome screen.

File: gaphor/ui/greeter.glade [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        <property name="can-focus">False</property>
+        <property name="label" translatable="yes">&lt;big&gt;&lt;b&gt;Welcome to Gaphor&lt;/b&gt;&lt;/big&gt;
+
+It's time to create your first model
+or open an existing model.
+
+Happy modeling!
+
+• &lt;a href="https://gaphor.org"&gt;Gaphor homepage&lt;/a&gt;
+• &lt;a href="https://github.com/gaphor"&gt;GitHub&lt;/a&gt;
```
Comment (danyeaw) [inline gaphor/ui/greeter.glade:99]: Did you mean to link this to the organization instead of the repo?

Comment (amolenaar) [reply gaphor/ui/greeter.glade:99]: Yes, that was intentional :)

**Pull request merged**